File: singly_linked_list/recipe-01/python/test_list_reverse.py
from singly_linked_list import SinglyLinkedList

def test():
    words = SinglyLinkedList()
    words.Append('foo')
    words.Append('bar')
    words.Append('bim')
    words.Append('baz')
    words.Append('quux')
    
    print("The origin list");
    for word in words:
        print("data: {}".format(word))

    print("Reverse the list");
    words.Reverse();
    for word in words:
        print("data: {}".format(word))

if __name__ == "__main__":
    test()

File: singly_linked_list/recipe-02/cxx/test_list_reverse.cpp
#include "singly_linked_list.hpp"
#include <string>
#include <iostream>

void test() {
    SinglyLinkedList<std::string> words;
    words.Append("foo");
    words.Append("bar");
    words.Append("bim");
    words.Append("baz");
    words.Append("quux");

    std::cout << "The origin list" << '\n';
    for (const auto& word : words)
        std::cout << "data: " << word << '\n';

    std::cout << "Reverse the list" << '\n';
    words.Reverse();
    for (const auto& word : words)
        std::cout << "data: " << word << '\n';
}

int main()
{
    test();
    return 0;
}


